test/parse/locator.test.js

```diff
@@ -28,76 +15,98 @@ describe('DOMLocator', () => {
 			'  -->',
 			'',
 			'  <state id="start" name="start">',
-			'    <transition event"init" name="init" target="main_state" />',
+			'    <transition event="init" name="init" target="main_state" />',
 			'  </state>',
 			'',
 			'  </scxml>',
 		].join('\n')
-		var parser = new DOMParser({ locator: {} })
-		var doc = parser.parseFromString(xml, 'text/xml')
-		var trans = doc.getElementsByTagName('transition')[0]
-		assert.equal(trans.lineNumber, 10) //,''+trans+trans.lineNumber+'/'+trans.parentNode.previousSibling.previousSibling.lineNumber)
+
+		const doc = new DOMParser().parseFromString(xml, 'text/xml')
+
+		expect(doc.getElementsByTagName('transition')[0]).toMatchObject({
+			lineNumber: 10,
```

Comment:
I think tests should be as specific as possible, so they the lowest amount of tests fail when something goes wrong.

This one is a good example: It verifies that empty lines are also counted by the locator.

Would we also check the columnNumber (which is covered by other tests) it would just fail together with other tests that verify them, but there is no need for it to fail of the line number for empty lines is still correct.

Reply:
Got it, ok.

Reply:
And maybe add a comment here?

Reply:
Done